widget/hexwidget/hexwidget.go

```diff
@@ -0,0 +1,240 @@
+	h.segmentObjects[5].(*canvas.Line).StrokeWidth = float32(hexSegmentWidth / 2)
+	h.segmentObjects[6].(*canvas.Line).StrokeWidth = float32(hexSegmentWidth / 2)
+
+	pos := image.Pt(0, 0)
+
+	pt0Center := image.Pt(pos.X+int(h.hex.hexWidth/2.0+h.hex.hexOffset), pos.Y)
+	pt05 := image.Pt(int(float32(pt0Center.X)-(hexSegmentHLength/2)), pt0Center.Y)
+	pt01 := image.Pt(int(float32(pt0Center.X)+(hexSegmentHLength/2)), pt0Center.Y)
+
+	pt6Center := image.Pt(pos.X+int(h.hex.hexWidth/2.0), int(float32(pt0Center.Y)+hexSegmentVLength))
+	pt65 := image.Pt(int(float32(pt6Center.X)-(hexSegmentHLength/2)), pt6Center.Y)
+	pt61 := image.Pt(int(float32(pt6Center.X)+(hexSegmentHLength/2)), pt6Center.Y)
+
+	pt3Center := image.Pt(pos.X+int(h.hex.hexWidth/2.0-h.hex.hexOffset), int(float32(pt0Center.Y)+2*hexSegmentVLength))
+	pt34 := image.Pt(int(float32(pt3Center.X)-(hexSegmentHLength/2)), pt3Center.Y)
+	pt32 := image.Pt(int(float32(pt3Center.X)+(hexSegmentHLength/2)), pt3Center.Y)
+
+	setLineEndpoints(h.segmentObjects[0].(*canvas.Line), pt05, pt01)
+	setLineEndpoints(h.segmentObjects[1].(*canvas.Line), pt01, pt61)
+	setLineEndpoints(h.segmentObjects[2].(*canvas.Line), pt61, pt32)
+	setLineEndpoints(h.segmentObjects[3].(*canvas.Line), pt32, pt34)
+	setLineEndpoints(h.segmentObjects[4].(*canvas.Line), pt34, pt65)
+	setLineEndpoints(h.segmentObjects[5].(*canvas.Line), pt65, pt05)
+	setLineEndpoints(h.segmentObjects[6].(*canvas.Line), pt65, pt61)
+
+	for i, v := range h.segmentObjects {
+		v.(*canvas.Line).StrokeColor = h.hex.getSegmentColor(i)
+		canvas.Refresh(v)
+	}
+}
+
+func (h *hexRenderer) Destroy() {
+}
+
+func (h *hexRenderer) Objects() []fyne.CanvasObject {
+	return h.segmentObjects
+}
+
+// HexWidget represents a 7-segment hexadecimal display. The segments
+// of the display mapped active-low onto 7 state bits, with segment 0 in
+// the least significant bit.
+//
+//       0
+//     -----
+//    |     |
+//  5 |     | 1
+//    |  6  |
+//     -----
+//    |     |
+//  4 |     | 2
+//    |  3  |
+//     -----
+type HexWidget struct {
+	widget.BaseWidget
+	segments uint8
+
+	// size in "pixels" of the hex widget
+	hexHeight float32
+	hexWidth  float32
+
+	// slant angle
+	hexOffset float32
+
+	// color when the hex is on
+	hexOnColor color.RGBA
+
+	// color when the hex is off
+	hexOffColor color.RGBA
+}
+
+// SetOnColor changes the color that segments are shown as when they are
+// active/on.
+func (h *HexWidget) SetOnColor(c color.RGBA) {
+	h.hexOnColor = c
+	h.Refresh()
+}
+
+// SetOffColor changes the color that segments are shown as when they are
+// inactive/off.
+func (h *HexWidget) SetOffColor(c color.RGBA) {
+	h.hexOffColor = c
+	h.Refresh()
+}
+
+// SetSize changes the size of the hex widget.
+func (h *HexWidget) SetSize(s fyne.Size) {
+	h.hexHeight = s.Height
+	h.hexWidth = s.Width
+	h.Refresh()
+}
+
+// SetSlant changes the amount of "slant" i nthe hex widgets. The topmost
+// segment is offset by slant many virtual pixels to the right. A value of 0
```

Comment:
Not sure that "virtual pixels" means much, sounds like pixels again.

Reply:
Also, for clarity, is it worth stating (if this is true?) that setting slant = height will give 45degree slope?

Reply:
AFAIK, this is correct.

bcf4e7c